mongodbatlas/resource_mongodbatlas_global_cluster_config.go

```diff
@@ -125,38 +123,41 @@ func resourceMongoDBAtlasGlobalClusterCreate(ctx context.Context, d *schema.Reso
 				addManagedNamespace.IsShardKeyUnique = pointy.Bool(isShardKeyUnique.(bool))
 			}
 
-			_, _, err := conn.GlobalClusters.AddManagedNamespace(ctx, projectID, clusterName, addManagedNamespace)
-
+			err := resource.RetryContext(ctx, 2*time.Minute, func() *resource.RetryError {
+				_, _, err := conn.GlobalClusters.AddManagedNamespace(ctx, projectID, clusterName, addManagedNamespace)
+				if err != nil {
+					switch {
+					case strings.Contains(fmt.Sprint(err), "DUPLICATE_MANAGED_NAMESPACE"):
```

Comment:
[nit] please let's stop doing `strings.Contains` for error checking is a bad practice and go1.13 introduced proper ways of doing this, https://go.dev/blog/go1.13-errors

Please use `errors.As` to cast to an error from the API client and check proper error code
```go
var target *matlas.ErrorResponse
if errors.As(err, &target) && target.ErrorCode == "DUPLICATE_MANAGED_NAMESPACE" {
...
```

Reply:
TIL!

Reply:
Interesting, understood and will make changes, thank you gustavo